### PR TITLE
Warn when Resource method returns JSON but no JSON provider exists

### LINKED ISSUE #22970: warning about missing json when just using resteasy or resteasy-reactive
### Describe the bug

when just using resteasy or resteasy-classic you should get a warning  that you are missing json support according to this code:

https://github.com/quarkusio/quarkus/blob/main/extensions/resteasy-classic/resteasy-common/deployment/src/main/java/io/quarkus/resteasy/common/deployment/ResteasyCommonProcessor.java#L220

But I get nothing. neither when using maven,gradle or jbang.

Attached is a resteasy classic reproducer. if change it to use resteasy-jackson I get json, but without it I get string and not warning.

[resteasy.zip](https://github.com/quarkusio/quarkus/files/7889504/resteasy.zip)

### Expected behavior

print during build of missing json provider.

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by geoand**
Resolves: #22970

### CODE REVIEW

File: extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/QuarkusServerEndpointIndexer.java [hunk 1]
```diff
@@ -125,4 +148,55 @@ public Builder setDefaultProducesHandler(DefaultProducesHandler defaultProducesH
...
+        super.handleAdditionalMethodProcessing(method, currentClassInfo, info, annotationStore);
+        warnAboutMissingJsonProviderIfNeeded(method, info);
+    }
+
+    private void warnAboutMissingJsonProviderIfNeeded(ServerResourceMethod method, MethodInfo info) {
+        if (!capabilities.isCapabilityWithPrefixMissing("io.quarkus.resteasy.reactive.json")) {
+            return;
+        }
+        if (hasJson(method) || isDefaultJson()) {
+            LOGGER.warnf("Quarkus detected the use of JSON is JAX-RS method '" + info.declaringClass().name() + "#"
```
Comment (gsmet) [inline extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/QuarkusServerEndpointIndexer.java:164]:
```suggestion
            LOGGER.warnf("Quarkus detected the use of JSON in JAX-RS method '" + info.declaringClass().name() + "#"
```

**Review by gsmet — DISMISSED** [review]
Spotted a small typo.

Don't we need the same for the client?

**Comment by geoand** [comment]
> Don't we need the same for the client?

Yeah, but that might need some extra changes

File: extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/QuarkusServerEndpointIndexer.java (hunk 1 above)
Comment (geoand) [reply extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/QuarkusServerEndpointIndexer.java]: Fixes

**Comment by gsmet** [comment]
CI is not happy about this:
```
Error: ]: Build step io.quarkus.resteasy.reactive.server.deployment.ResteasyReactiveProcessor#setupEndpoints threw an exception: java.lang.RuntimeException: java.lang.RuntimeException: Failed to process method 'io.quarkus.resteasy.reactive.server.test.customproviders.UniFiltersResource#filters'
	at org.jboss.resteasy.reactive.common.processor.EndpointIndexer.createEndpoints(EndpointIndexer.java:278)
	at io.quarkus.resteasy.reactive.server.deployment.ResteasyReactiveProcessor.setupEndpoints(ResteasyReactiveProcessor.java:525)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:882)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
	at java.base/java.lang.Thread.run(Thread.java:829)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
Caused by: java.lang.RuntimeException: Failed to process method 'io.quarkus.resteasy.reactive.server.test.customproviders.UniFiltersResource#filters'
	at org.jboss.resteasy.reactive.common.processor.EndpointIndexer.createResourceMethod(EndpointIndexer.java:639)
	at org.jboss.resteasy.reactive.common.processor.EndpointIndexer.createEndpoints(EndpointIndexer.java:366)
	at org.jboss.resteasy.reactive.common.processor.EndpointIndexer.createEndpoints(EndpointIndexer.java:249)
	... 12 more
Caused by: java.lang.NullPointerException
	at io.quarkus.resteasy.reactive.common.deployment.JsonDefaultProducersHandler.handle(JsonDefaultProducersHandler.java:26)
	at io.quarkus.resteasy.reactive.server.deployment.QuarkusServerEndpointIndexer.isDefaultJson(QuarkusServerEndpointIndexer.java:171)
	at io.quarkus.resteasy.reactive.server.deployment.QuarkusServerEndpointIndexer.warnAboutMissingJsonProviderIfNeeded(QuarkusServerEndpointIndexer.java:163)
	at io.quarkus.resteasy.reactive.server.deployment.QuarkusServerEndpointIndexer.handleAdditionalMethodProcessing(QuarkusServerEndpointIndexer.java:156)
	at io.quarkus.resteasy.reactive.server.deployment.QuarkusServerEndpointIndexer.handleAdditionalMethodProcessing(QuarkusServerEndpointIndexer.java:32)
	at org.jboss.resteasy.reactive.common.processor.EndpointIndexer.createResourceMethod(EndpointIndexer.java:631)
	... 14 more

	at io.quarkus.test.QuarkusUnitTest.beforeAll(QuarkusUnitTest.java:613)
```

**Comment by geoand** [comment]
Should be an easy fix. I'll fix it in the morning

**Review by gsmet — APPROVED** [review]

**Review by mamadaliev — APPROVED** [review]

**Pull request merged**